python/spacewalk/satellite_tools/reposync.py

```diff
@@ -751,10 +752,16 @@ def sync(self, update_repodata=True):
                         # this fetch also the normal xml primary file
                         repo_checksum_type = plugin.get_md_checksum_type()
 
+                        modulemd_importer = None
                         if not self.no_packages:
                             self.import_groups(plugin)
                             if repo_type == "yum":
-                                self.import_modules(plugin)
+                                modulemd_path = self.import_modules(plugin)
+                                if modulemd_path:
+                                    modulemd_importer = ModuleMdImporter(
+                                        self.channel["id"], modulemd_path
+                                    )
+                                    modulemd_importer.validate()
```

Comment:
This can raise an exception, can you add code that handles it? The general exception handler logs with "Unexpected error", maybe we can use a concrete logging statement instead.